presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/environment/SinglenodeKafka.java

```diff
@@ -0,0 +1,87 @@
+import io.prestosql.tests.product.launcher.env.Environment;
+import io.prestosql.tests.product.launcher.env.common.AbstractEnvironmentProvider;
+import io.prestosql.tests.product.launcher.env.common.Standard;
+
+import javax.inject.Inject;
+
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_CATALOG;
+import static io.prestosql.tests.product.launcher.env.common.Hadoop.CONTAINER_PRESTO_KAFKA_PROPERTIES;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public class SinglenodeKafka
+        extends AbstractEnvironmentProvider
+{
+    private static final String CONFLUENT_VERSION = "5.2.1";
+
+    private final DockerFiles dockerFiles;
+
+    @Inject
+    public SinglenodeKafka(Standard standard, DockerFiles dockerFiles)
+    {
+        super(ImmutableList.of(standard));
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+    }
+
+    @Override
+    protected void extendEnvironment(Environment.Builder builder)
+    {
+        super.extendEnvironment(builder);
+
+        builder.configureContainer("presto-master", container -> container
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka.properties"), CONTAINER_PRESTO_KAFKA_PROPERTIES, READ_ONLY))
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka"), CONTAINER_PRESTO_CATALOG + "kafka", READ_ONLY));
+
+        builder.configureContainer("tests", container -> container
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("conf/environment/singlenode-kafka/kafka"), CONTAINER_PRESTO_CATALOG + "kafka", READ_ONLY));
+
+        builder.addContainer("zookeeper", createZookeeper());
+        builder.addContainer("kafka", createKafka());
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createZookeeper()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-zookeeper:" + CONFLUENT_VERSION)
+                .withNetworkAliases("zookeeper")
+                .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+                .withEnv("ZOOKEEPER_TICK_TIME", "2000")
+                .withExposedPorts(2181);
+        container.setDockerImageName("zookeeper");
+        return container;
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createKafka()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-kafka:" + CONFLUENT_VERSION)
+                .withNetworkAliases("kafka")
+                .withEnv("KAFKA_BROKER_ID", "1")
+                .withEnv("KAFKA_ZOOKEEPER_CONNECT", "zookeeper:2181")
+                .withEnv("KAFKA_ADVERTISED_LISTENERS", "PLAINTEXT://kafka:9092")
+                .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1")
+                .withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0")
+                .withExposedPorts(9092);
```

Comment:
the port gets exposed on localhost at random number

that's unlikely useful. Skip it?
Or do you use it as an implicit health check? (yes =>  add a comment)